crates/fluvio-version-manager/src/command/switch.rs

```diff
@@ -0,0 +1,230 @@
+        }
+
+        let fluvio_bin_dir = fluvio_binaries_path()?;
+
+        if !fluvio_bin_dir.exists() {
+            tracing::info!(?fluvio_bin_dir, "Creating fluvio binaries directory");
+            create_dir_all(&fluvio_bin_dir)?;
+        }
+
+        // Opens the version directory and then copies the contents (but the manifest)
+        // to the Fluvio Binaries directory
+        let version_dir = VersionDirectory::open(pkgset_path)?;
+        version_dir.copy_contents(&fluvio_bin_dir)?;
+
+        // Update settings file to keep track of active Fluvio Version
+        let manifest = VersionManifest::open(version_dir.manifest)?;
+        let mut settings = Settings::open()?;
+        settings.update_version(&manifest.channel, &manifest.version)?;
+
+        self.notify_done(format!(
+            "Now using Fluvio version {}",
+            version.to_string().bold(),
+        ));
+
+        Ok(())
+    }
+}
+
+impl Notify for SwitchOpt {
+    fn is_quiet(&self) -> bool {
+        self.global_opts.quiet
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::fs::File;
+
+    use super::*;
+
+    #[test]
+    fn opens_directory_as_version_dir() {
+        let tmpdir = tempfile::tempdir().unwrap();
+        let manifest = tmpdir.path().join(PACKAGE_SET_MANIFEST_FILENAME);
+        let random_file = tmpdir.path().join("random_file");
+
+        File::create(&manifest).unwrap();
+        File::create(&random_file).unwrap();
+
+        let entries = VersionDirectory::open(tmpdir.path().to_path_buf()).unwrap();
+
+        assert_eq!(entries.contents.len(), 1);
+        assert_eq!(entries.contents[0], random_file);
+        assert_eq!(entries.manifest, manifest);
+    }
+
+    #[test]
+    fn fails_to_open_directory_without_manifest() {
+        let tmpdir = tempfile::tempdir().unwrap();
+        let result = VersionDirectory::open(tmpdir.path().to_path_buf());
+
+        assert!(result.is_err());
+        assert_eq!(
+            result.err().unwrap().to_string(),
+            "Failed to find manifest file in version directory"
+        );
+    }
+
+    #[test]
+    fn copies_contents_to_target_directory() {
+        let tmpdir = tempfile::tempdir().unwrap();
+        let target_dir = tempfile::tempdir().unwrap();
+        let manifest = tmpdir.path().join(PACKAGE_SET_MANIFEST_FILENAME);
+        let random_file = tmpdir.path().join("random_file");
+
+        File::create(manifest).unwrap();
+        File::create(&random_file).unwrap();
+
+        let entries = VersionDirectory::open(tmpdir.path().to_path_buf()).unwrap();
+        entries.copy_contents(&target_dir).unwrap();
+
+        let target_entries = read_dir(target_dir.path()).unwrap();
+        let target_entries: Vec<PathBuf> =
+            target_entries.map(|entry| entry.unwrap().path()).collect();
+
+        assert_eq!(target_entries.len(), 1);
+        assert_eq!(
+            target_entries.get(0).unwrap(),
+            &target_dir.path().join(random_file.file_name().unwrap())
```

Comment:
Need to confirm copies are done by checking file contents and they are same

Reply:
I think I could generate SHA256 checksums from each file and compare as part of the tests!